### DISCUSSION TITLE
Have the cursor always be under the window

### DISCUSSION THREAD

**Discussion by FilipRank** [rxhanson/Rectangle#822] [Ideas]
When I have a window fully expanded on my desktop and drag the window down so that it goes back to its original size, the window is now no longer under the cursor, but of to its side while it remains dragable. Combine this with the fact that the people at Mac HQ thought that it would be a good idea to be able to drag your windows off of the desktop, and this can be a problem. If the cursor always stayed above the window when dragging, it could help.

**Comment by rxhanson** [comment]
Technically, it should be possible to move the cursor when the window restores, but there could be some weirdness there with the drag started. It could also be undesirable for some users to have their mouse cursor move to a different location, too. Because of that, I don't have a strong desire to make that change, but I see no reason to turn down a good pull request that allows this behavior to be enabled via Terminal command.

As an aside, I have looked into making the window move so that it stays under the cursor (maybe Windows does this, if I remember right?), but it doesn't work properly with what's available to me in macOS.

**Reply by Aaron-Rumpler** [reply]
Shame macOS doesn't let you move a window that's being dragged. Would have been the ideal solution.

**Reply by Aaron-Rumpler** [reply]
As for moving the cursor, I suspect you'll have the same issue. It looks to me like macOS keeps track of where on the window you started the drag, and then whenever the mouse moves, it moves the window so that point is under the cursor. By moving the cursor programmatically, I suspect the window will move with it. Still worth a try if someone wants to though, as I might be wrong.

**Comment by t3chguy** [comment]
So I am experiencing this same issue and it does not occur with Magnet, where the window is moved during the unsnap/restore/drag to be positioned under your cursor.

Rectangle:

https://user-images.githubusercontent.com/2403652/225898118-0966f17a-ea9f-4e78-86b5-8fb681c028dd.mov

Magnet:

https://user-images.githubusercontent.com/2403652/225898662-53d3978a-0e6a-46e1-9214-75721fa2c5f0.mov

**Reply by rxhanson** [reply]
Thanks! Looks like things have changed a bit in macOS to allow this to work. I'll see what I can do.
